### ISSUE TITLE
Wrong ':' token evaluation in arrays

### ISSUE THREAD

**Issue by emaincourt** [eslint/eslint#9139]
**Tell us about your environment**

* **ESLint Version: 3.19.0**
* **Node Version: 6.10**
* **npm Version: 3.10.10**

**What parser (default, Babel-ESLint, etc.) are you using?**

Babel-ESLint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parser: 'babel-eslint',
  extends: 'airbnb',
  env: {
    node: true,
    jest: true,
    jasmine: true,
  },
  rules: {
    'no-console': 'error',
    strict: 'off',
    'import/prefer-default-export': 'off',
    'jsx-a11y/anchor-is-valid': 'off',
    'jsx-a11y/href-no-hash': 'off',
  },
};
```
</details>

**What did you do? Please include the actual source code causing the issue.**

```js
const foo = 'foo' : 'bar'; // Error raised
const bar = [
  'foo' : 'bar', // No error raised
];
```

**What did you expect to happen?**

I was expecting the `bar` declaration to raise an error since `'foo' : 'bar'` doesn't make, as far as I know, any sense outside of a ternary expression scope.

**What actually happened? Please include the actual, raw output from ESLint.**

This led to an inability for babel to transpile the code. Is it an expected behaviour ? And if yes, what is it's purpose ?

**Comment by platinumazure** [comment]
I'm not sure this is an ESLint issue. If that expression should fail to parse (and I think it should), that is babel-eslint's responsibility to do so before the file gets to ESLint for linting.

You might want to file an issue at babel-eslint instead: https://github.com/babel/babel-eslint/issues/new

**Comment by emaincourt** [comment]
Hi @platinumazure,

Thank you for the answer. It makes sense, I'll so forward the issue to babel.

**Comment by not-an-aardvark** [comment]
Closing because this isn't actionable on ESLint's end (it's presumably a `babel-eslint` bug).
